contracts/src/AccountRegistry.sol

```diff
@@ -218,27 +232,36 @@ contract AccountRegistry is Initializable, EIP712Upgradeable, Ownable2StepUpgrad
     }
 
     /**
-     * @dev Helper function to get pubkey bitmap from packed
+     * @dev Helper function to get pubkey bitmap from the packed storage.
      */
     function _getPubkeyBitmap(uint256 accountIndex) internal view returns (uint256) {
-        return _accountIndexToRecoveryAddressPacked[accountIndex] >> 160;
+        return uint32(_accountIndexToRecoveryAddressPacked[accountIndex] >> 224);
     }
 
     /**
-     * @dev Helper function to set pubkey bitmap in packed
+     * @dev Helper function to set pubkey bitmap packed, preserving the recovery address. The
+     * bitmap is 32 bits, but 256 are accepted to simplify bit operations in other functions.
      */
     function _setPubkeyBitmap(uint256 accountIndex, uint256 bitmap) internal {
+        if (bitmap >> 224 != 0) {
```

Comment:
Don't you want to check here that only the lowest 32 bit are set? i.e. `bitmap >> 32 != 0`